src/script/src/python/ffi_types/copr.rs

```diff
@@ -414,30 +414,38 @@ impl PyQueryEngine {
             .map_err(|e| format!("Dedicated thread for sql query panic: {e:?}"))?
     }
     // TODO(discord9): find a better way to call sql query api, now we don't if we are in async context or not
-    /// return sql query results in List[List[PyVector]], or List[usize] for AffectedRows number if no recordbatches is returned
+    /// return sql query results in List[PyVector], or List[usize] for AffectedRows number if no recordbatches is returned
     #[pymethod]
     fn sql(&self, s: String, vm: &VirtualMachine) -> PyResult<PyListRef> {
         self.query_with_new_thread(s)
             .map_err(|e| vm.new_system_error(e))
             .map(|rbs| match rbs {
                 Either::Rb(rbs) => {
-                    let mut top_vec = Vec::with_capacity(rbs.iter().count());
+                    let mut columns_vectors: Vec<PyVectorRef> = Vec::new();
                     for rb in rbs.iter() {
-                        let mut vec_of_vec = Vec::with_capacity(rb.columns().len());
-                        for v in rb.columns() {
-                            let v = PyVector::from(v.clone());
-                            vec_of_vec.push(v.to_pyobject(vm));
+                        for (i, v) in rb.columns().iter().enumerate() {
+                            let v = PyVector::from(v.clone()).into_ref(vm);
+                            match columns_vectors.get_mut(i) {
+                                Some(vector) => {
+                                    *vector = PyVector::concat_vector(vector, v, vm)?.into_ref(vm);
+                                }
```

Comment:
arrow provides a [method](https://docs.rs/arrow/latest/arrow/compute/kernels/concat/fn.concat_batches.html) to concat `RecordBatch`es which looks more efficient. Can we make use of it?

Reply:
Good suggestion!